src/channel/ChannelContext.cpp

```diff
@@ -255,7 +255,15 @@ void ChannelContext::HandleNodeIdResolve(CHIP_ERROR error, uint64_t nodeId, cons
 CHIP_ERROR ChannelContext::SendSessionEstablishmentMessage(const PacketHeader & header, const Transport::PeerAddress & peerAddress,
                                                            System::PacketBufferHandle msgIn)
 {
-    return mExchangeManager->GetSessionMgr()->GetTransportManager()->SendMessage(header, peerAddress, std::move(msgIn));
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    ReturnErrorOnFailure(header.EncodeBeforeData(msgIn));
+    ReturnErrorOnFailure(err);
```

Comment:
This line isn't needed.

Reply:
ChannelContext::SendSessionEstablishmentMessage call the low level transport directly to send the message instead of SecureSessionMgr.

err = mExchangeManager->GetSessionMgr()->GetTransportManager()->SendMessage(peerAddress, std::move(msgIn));

So we still need to encode the PacketHeader here before sending

Reply:
Acked. I think you just means line 261